mlops_utilities/helpers.py

```diff
@@ -98,69 +152,104 @@ def convert_param_dict_to_key_value_list(arg_dict: Dict[str, str]) -> List[Dict[
     :param arg_dict: key-value need to convert to AWS resource tags structure
     :return: list of tags in the following format: [ { "Key": "...", "Value": "..." }, ... ]
     """
-    return [
-        {'Key': k, 'Value': v}
-        for k, v in arg_dict.items()
-    ]
+    return [{"Key": k, "Value": v} for k, v in arg_dict.items()]
 
 
 def get_value_from_dict(data_dict: Dict[str, Any], path: List[str]) -> Mapping:
     """
-       Get necessary metric from evaluation.json
-       :param data_dict: dictionary that contains metrics {"regression_metrics": {"mse":..}}
-       :param path: path to metric ['regression_metrics','mse',…]
-       :return: dictionary that contains target metric:
-       for example {mse: 2.1} --from--> { "Key 01": { "Key 02:…{mse: 2.1}"}
+    Get necessary metric from evaluation.json
+    :param data_dict: dictionary that contains metrics {"regression_metrics": {"mse":..}}
+    :param path: path to metric ['regression_metrics','mse',…]
+    :return: dictionary that contains target metric:
+    for example {mse: 2.1} --from--> { "Key 01": { "Key 02:…{mse: 2.1}"}
     """
     return reduce(operator.getitem, path, data_dict)
 
 
 def normalize_key(key: str) -> str:
-    return key.replace('_', '').lower()
+    """
+    Remove _ from keys and lower case it
+    :param key:
+    :return key:
+    """
+    return key.replace("_", "").lower()
 
 
 def get_model_name(model_arn: str) -> str:
-    return model_arn[model_arn.rindex('/') + 1:]
+    """
+    Get model name from ARN
+    :param model_arn:
+    :return model name:
+    """
+    return model_arn[model_arn.rindex("/") + 1 :]
 
 
 def get_job_name(job_arn: str) -> str:
-    return job_arn[job_arn.rindex('/') + 1:]
+    """
+    Get job name from ARN
+    :param job_arn:
+    :return job name:
+    """
+    return job_arn[job_arn.rindex("/") + 1 :]
 
 
 def _list_to_dict(arg_list: List[Dict[str, Any]], dict_key_attr: str) -> Dict[Any, Dict]:
-    return {
-        o[dict_key_attr]: o
-        for o in arg_list
-    }
+    """
+    List to Dict
+    :param arg_list:
+    :param dict_key_attr:
+    :return:
+    """
+    return {o[dict_key_attr]: o for o in arg_list}
 
 
 def ensure_min_length(argument: str, min_length: int) -> str:
+    """
+    Add 0's to fulfill min length
```

Comment:
But why? 🤔

Reply:
good question

Reply:
Looks like legacy to hotfix bug in sagemaker API

Reply:
I can't found usage.